### PR TITLE
Getting started k8s part3 (resubmit PR)

### PR DESCRIPTION
**Pull request by spara**
Mike, my local repo got borked - resubmitting the PR with the changes you requested in the previous PR.

### CODE REVIEW

File: content/blog/getting-started-with-k8s-part3/index.md [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+<!--more-->
+
+This series walks you through:
+
+- [Building a Kubernetes cluster on cloud providers]({{< relref "/blog/getting-started-with-k8s-part1" >}})
+- [Basic application deployment]({{< relref "/blog/getting-started-with-k8s-part2" >}})
+- Advance application deployment and Helm charts
+- Stateful applications
+- Networking
+- “Day 2” activities such as migrating node groups.
```
Comment (metral) [inline content/blog/getting-started-with-k8s-part3/index.md:24]: We also have this already: https://www.pulumi.com/docs/tutorials/kubernetes/eks-migrate-nodegroups/

File: content/blog/getting-started-with-k8s-part3/index.md [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+
+In this article, we’ll assume that we already have a Kubernetes cluster available. You will have to set the [context](https://www.pulumi.com/docs/intro/cloud-providers/kubernetes/setup/#kubernetes-configuration) with *kubectl* to deploy the examples in this article. Because of Kubernetes’ OpenAPI, we can use the same Typescript application code on any compliant instance.
+
+## Sock Shop Microservices
+
+We’ll use the [Sock Shop](https://github.com/pulumi/examples/tree/master/kubernetes-ts-sock-shop) microservices reference application. This is a polyglot application comprised of microservices written in Java, Go, and Node. This demo application adheres to the 12-factor app pattern by creating a microservice for each function. The application services will be deployed to pods, and they communicate with each other via REST calls over HTTP.
+
+![Architecture]()
```
Comment (metral) [inline content/blog/getting-started-with-k8s-part3/index.md:34]: Image is missing

File: content/blog/getting-started-with-k8s-part3/index.md [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+
+Although we encapsulate the deployment in a single large script, we can improve on that by creating components using classes to abstract the implementation details.
+
+## Deploying Guestbook as a Component
+
+The Kubernetes [GuestBook](https://kubernetes.io/docs/tutorials/stateless-application/guestbook/) is a canonical example of a stateless application. It consists of multiple PHP frontend services backed by Redis replicas and a lead Redis database to store guestbook entries. In the previous example, we created a Deployment and a Service for each microservice in the application. Although this is a valid pattern for deploying an application in Kubernetes, it doesn’t take full advantage of the capabilities of infrastructure as code.
+
+We can abstract objects, such as Deployments and Services, into components to create our own resources. [ComponentResource]({{< relref "/docs/reference/pkg/nodejs/pulumi/pulumi#ComponentResource" >}}) is Pulumi construct or resource that can aggregate other resources into a combined resource. Let’s take a look at how we implement the [Guestbook with components](https://github.com/pulumi/examples/tree/master/kubernetes-ts-guestbook/components).
```
Comment (metral) [inline content/blog/getting-started-with-k8s-part3/index.md:224]: Consider moving this paragraph below the code snippet just below it, but before the `ComponentResource` example

Comment (spara) [reply content/blog/getting-started-with-k8s-part3/index.md:224]: accepted

File: content/blog/getting-started-with-k8s-part3/index.md (hunk 2 above)
Comment (spara) [reply content/blog/getting-started-with-k8s-part3/index.md:34]: image added

File: content/blog/getting-started-with-k8s-part3/index.md (hunk 1 above)
Comment (spara) [reply content/blog/getting-started-with-k8s-part3/index.md:24]: I plan to use it as an example

**Pull request merged**